Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm

```diff
@@ -501,15 +501,22 @@ static ASCIILiteral serviceName(const ProcessLauncher::LaunchOptions& launchOpti
 
 void ProcessLauncher::terminateProcess()
 {
+#if USE(EXTENSIONKIT)
+    if (m_process)
+        m_process->invalidate();
+#endif
+
     if (m_isLaunching) {
         terminateXPCConnection();
         return;
     }
 
-    if (!m_processID)
-        return;
+#if !PLATFORM(COCOA)
```

Comment:
We're in a Cocoa.mm file so this check doesn't make much sense.. It's essentially dead code after your change.

Reply:
Ah, these are all good point. I will address in the upcoming patch.

Thanks for reviewing!